perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2984,120 +3183,919 @@ private void closeExtSourcesAfterSynchronization(ExtSource membersSource, ExtSou
+					skippedGroups.add("GroupEntry:[" + groupToUpdate + "] was skipped because group move is not allowed: Exception: " + e.getName() + " => " + e.getMessage() + "]");
+					continue;
+				} catch (WrongAttributeValueException e) {
+					log.warn("Can't update group {} from baseGroup {} due to wrong attribute value exception {}.", groupToUpdate, e);
+					skippedGroups.add("GroupEntry:[" + groupToUpdate + "] was skipped because wrong attribute value: Exception: " + e.getName() + " => " + e.getMessage() + "]");
+					continue;
+				} catch (WrongReferenceAttributeValueException e) {
+					log.warn("Can't update group {} from baseGroup {} due to wrong reference attribute value exception {}.", groupToUpdate, e);
+					skippedGroups.add("GroupEntry:[" + groupToUpdate + "] was skipped because wrong reference attribute value: Exception: " + e.getName() + " => " + e.getMessage() + "]");
+					continue;
+				}
+			}
+
+			boolean changed = updateGroupDescription(sess, groupToUpdate, candidateGroup);
+			if(changed) {
+				log.trace("Group structure synchronization {}: value of the group description for groupId {} changed. Original value {}, new value {}.",
+						baseGroup, groupToUpdate.getId(), groupToUpdate.getDescription(), candidateGroup.asGroup().getDescription());
+			}
+
+		}
+	}
+
+	/**
+	 * Specify parent group for group which is updated by synchronization
+	 *
+	 * 1. Gets actual parent group name
+	 * 2. If candidate group doesn't have parent and actual parent is not base group, specify base group as parent
+	 * 3. if both groups (candidate and toUpdate) have parents which are not equal there can be two scenarios:
+	 *		3.1 There is subGroup with name equals to candidateGroups parent name -> specify subGroup as new parent
+	 *		3.2 There isn't subGroup with name equals to candidateGroup parent name -> specify baseGroup as new parent
+	 * 4. In case parent group hasn't changed, method will return null
+	 *
+	 * Method is used by group structure synchronization.
+	 *
+	 * @param groupToUpdate for which will be parent specified
+	 * @param baseGroup under which may be new parent
+	 * @param candidateGroup with potential new parent group
+	 * @return updated parent group (if null, parent group hasn't changed)
+	 * @throws InternalErrorException
+	 */
+	private Group specifyParentForUpdatedGroup(PerunSession sess, Group groupToUpdate, Group baseGroup, CandidateGroup candidateGroup) throws InternalErrorException {
+
+		String actualParentName;
+		try {
+			actualParentName = getGroupById(sess, groupToUpdate.getParentGroupId()).getShortName();
+		} catch (GroupNotExistsException e) {
+			throw new InternalErrorException(e);
+		}
+
+		Group newParentGroup = null;
+
+		if(candidateGroup.getParentGroupName() == null){
+			if(!actualParentName.equals(baseGroup.getShortName())){
+				newParentGroup = baseGroup;
+			}
+		}
+
+		else if(!actualParentName.equals(candidateGroup.getParentGroupName())){
```

Comment:
This is not the best way how to write else if. You should continue where first part of if is ending. This is confusing for reading.
if { ...
} else if {
...
} else {
...
}